### PR TITLE
docs - sql ref page updates for CHECKPOINT, CLUSTER, COMMENT

### PR DESCRIPTION
**Pull request by lisakowen**
sql ref page updates for CHECKPOINT, CLUSTER, COMMENT to bring them up to postgres 12 content.  there are some greenplum-specific statements on the CLUSTER page.

CLUSTER
- some greenplum-specific info, tagged with XXX.  is this still valid in greenplum 7?

COMMENT:
- omitted LARGE OBJECT, PUBLICATION, SUBSCRIPTION

not in this PR:  there is some great info in this cluster blog (https://greenplum.org/gp7-clustering-aoco-tables/).  created a story to incorporate it into the narrative docs.

doc review site links (behind vpn), can navigate to all ref pages from here:  https://docs-staging.vmware.com/en/draft/VMware-Greenplum/sqlref-ccc/greenplum-database/GUID-ref_guide-sql_commands-CHECKPOINT.html

### CODE REVIEW

File: gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md [hunk 1]
```diff
@@ -5,32 +5,38 @@ Physically reorders a table on disk according to an index.
...
-CLUSTER [VERBOSE] <tablename> [ USING index_name ]
+CLUSTER [VERBOSE] <table_name> [ USING <index_name> ]
 
 CLUSTER [VERBOSE]
 ```
 
 ## <a id="section3"></a>Description 
 
-`CLUSTER` orders a table based on an index. `CLUSTER` on non-B-tree indexes is not supported on append-optimized storage tables. Clustering an index means that the records are physically ordered on disk according to the index information. If the records you need are distributed randomly on disk, then the database has to seek across the disk to get the records requested. If those records are stored more closely together, then the fetching from disk is more sequential. A good example for a clustered index is on a date column where the data is ordered sequentially by date. A query against a specific date range will result in an ordered fetch from the disk, which leverages faster sequential access.
+`CLUSTER` instructs Greenplum Database to order the table specified by `<table_name>` based on the index specified by `<index_name>`. The index must already have been defined on `<table_name>.
```
Comment (Chibin) [inline gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md:17]:
```suggestion
`CLUSTER` instructs Greenplum Database to order the table specified by `<table_name>` based on the index specified by `<index_name>`. The index must already have been defined on `<table_name>`.
```

File: gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md [hunk 2]
```diff
@@ -5,32 +5,38 @@ Physically reorders a table on disk according to an index.
...
+`CLUSTER` instructs Greenplum Database to order the table specified by `<table_name>` based on the index specified by `<index_name>`. The index must already have been defined on `<table_name>`.
+
+XXX `CLUSTER` on non-B-tree indexes is not supported on append-optimized storage tables. XXX
+
+> **Note** Greenplum Database 7 does not support `CLUSTER`ing a partitioned table.
+
+When a table is clustered, it is physically reordered on disk based on the index information. XXX If the records you need are distributed randomly on disk, then the database has to seek across the disk to get the records requested. If those records are stored more closely together, then the fetching from disk is more sequential. A good example for a clustered index is on a date column where the data is ordered sequentially by date. A query against a specific date range will result in an ordered fetch from the disk, which leverages faster sequential access. XXX
```
Comment (Chibin) [inline gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md:23]: The XXX portion seems to be very similar to the first paragraph of the `Notes`.

Perhaps combining them would be good?
Thoughts?

File: gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md [hunk 3]
```diff
@@ -43,13 +49,13 @@ When an index scan is used, a temporary copy of the table is created that contai
...
+It is advisable to set the [maintenance\_work\_mem](../config_params/guc-list.html) configuration parameter to a reasonably large value \(but not more than the amount of RAM you can dedicate to the `CLUSTER` operation\) before clustering.
 
-Because the query optimizer records statistics about the ordering of tables, it is advisable to run `ANALYZE` on the newly clustered table. Otherwise, the planner may make poor choices of query plans.
+Because the query optimizer records statistics about the ordering of tables, it is advisable to run [ANALYZE](ANALYZE.html) on the newly clustered table. Otherwise, the planner may make poor query plan choices.
 
 Because `CLUSTER` remembers which indexes are clustered, you can cluster the tables you want clustered manually the first time, then set up a periodic maintenance script that runs `CLUSTER` without any parameters, so that the desired tables are periodically reclustered.
 
-> **Note** `CLUSTER` on non-B-tree indexes is not supported with append-optimized tables.
+> **Note** XXX `CLUSTER` on non-B-tree indexes is not supported with append-optimized tables.
```
Comment (Chibin) [inline gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md:58]: Should we just say:
CLUSTER using B-tree indexes is supported for append-optimized tables?

File: gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md [hunk 4]
```diff
@@ -59,6 +65,20 @@ Cluster the table `employees` on the basis of its index `emp_ind`:
...
+CLUSTER employees;
+```
+
+Cluster all tables in the database that have previously been clustered:
+
+```
+CLUSTER;
+```
+
+XXX
```
Comment (Chibin) [inline gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md:80]: Cluster a large table by recreating it and loading it in the correct index order

This is no longer needed. If the table already has an index, we can just call `CLUSTER`.

File: gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md [hunk 5]
```diff
@@ -5,32 +5,38 @@ Physically reorders a table on disk according to an index.
...
 
 CLUSTER [VERBOSE]
 ```
 
 ## <a id="section3"></a>Description 
 
-`CLUSTER` orders a table based on an index. `CLUSTER` on non-B-tree indexes is not supported on append-optimized storage tables. Clustering an index means that the records are physically ordered on disk according to the index information. If the records you need are distributed randomly on disk, then the database has to seek across the disk to get the records requested. If those records are stored more closely together, then the fetching from disk is more sequential. A good example for a clustered index is on a date column where the data is ordered sequentially by date. A query against a specific date range will result in an ordered fetch from the disk, which leverages faster sequential access.
+`CLUSTER` instructs Greenplum Database to order the table specified by `<table_name>` based on the index specified by `<index_name>`. The index must already have been defined on `<table_name>`.
+
+XXX `CLUSTER` on non-B-tree indexes is not supported on append-optimized storage tables. XXX
```
Comment (Chibin) [inline gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md:19]: We mention this XXX non-B-tree twice in this page. Is that okay?

File: gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md (hunk 3 above)
Comment (lisakowen) [reply gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md:58]: yeah, the original text is kind of interesting.  i can reword.  but also wondering if we support clustering on other types of indexes for append-optimized tables?

**Comment by lisakowen** [comment]
thank you for reviewing, @Chibin .  the original writer chose to duplicate some of the info, not sure why.  i will try to combine to a single ref.

File: gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md (hunk 2 above)
Comment (lisakowen) [reply gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md:23]: moved the info to the notes section.

File: gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md (hunk 3 above)
Comment (Chibin) [reply gpdb-doc/markdown/ref_guide/sql_commands/CLUSTER.html.md:56]: From what I can see in the code, we currently only support b-tree indexes for append-optimized tables.

**Review by Chibin — APPROVED** [review]
LGTM!

**Comment by lisakowen** [comment]
thank you for the thorough review, @Chibin !

**Review by dyozie — APPROVED** [review]

**Pull request merged**